parser/binary-parser_test.go

```diff
@@ -0,0 +1,117 @@
+package parser
+
+import (
+	"encoding/json"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+
+	tfjson "github.com/hashicorp/terraform-json"
+)
+
+// MockCommandExecutor mocks the command execution.
+type MockCommandExecutor struct {
+	Output     []byte
+	Err        error
+	ActualName string
+	ActualArgs []string
+}
+
+func (e *MockCommandExecutor) CombinedOutput(name string, args ...string) ([]byte, error) {
```

Comment:
Shall we use gomock to generate the mocks?

Reply:
Sure, will do that!